### PR TITLE
fix(Container): Added comprobation to edge cases

### PR DESCRIPTION
**Pull request by xxxtonixxx**
Thanks for your awesome library, it is really a good job! 👍 

It allows to set 0, false and "" (empty string) like value.

should be `null` an allowed value?

### CODE REVIEW

**Comment by pleerock** [comment]
> should be null an allowed value?

not sure TBH. Let's keep it as it until someone request it

**Comment by pleerock** [comment]
I added a strict null check, why it failed?

**Comment by xxxtonixxx** [comment]
Because if `service.instance` is `undefined` it's returned. `service.instance != null` checks both (`null` and `undefined`, but not empty string, 0 or false)

File: src/Container.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class Container {
         let service = this.findService(identifier);
 
         // find if instance of this object already initialized in the container and return it if it is
-        if (service && service.instance !== null)
+        if (service && service.instance !== null && service.instance !== undefined)
```
Comment (xxxtonixxx) [inline src/Container.ts]: why don't symply `service.instance != null`?

Comment (pleerock) [reply src/Container.ts]: because of strict checking rule

**Pull request merged**